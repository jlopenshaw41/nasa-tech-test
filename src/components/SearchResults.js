import React from "react";
import "../styles/SearchResults.css";
import PropTypes from "prop-types";

const SearchResults = ({ results }) => {
  return results.length ? (
    results.map((result) => {
      return (
        <>
          <img
            className="image"
            src={result}
            alt="Search result from NASA database"
            data-testid="image"
          />
        </>
      );
    })
  ) : (
    <p>No results</p>
  );
};

SearchResults.propTypes = {
  results: PropTypes.array.isRequired,
}

export default SearchResults;
