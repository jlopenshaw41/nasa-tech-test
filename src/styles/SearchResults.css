.search-results > img {
    height: 150px;
}