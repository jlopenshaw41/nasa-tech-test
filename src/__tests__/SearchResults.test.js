import { render } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("<SearchResults />", () => {
  const images = [
    "https://images-assets.nasa.gov/image/PIA06632/PIA06632~thumb.jpg",
    "https://images-assets.nasa.gov/image/PIA06640/PIA06640~thumb.jpg",
    "https://images-assets.nasa.gov/image/PIA07606/PIA07606~thumb.jpg",
    "https://images-assets.nasa.gov/image/PIA08946/PIA08946~thumb.jpg",
    "https://images-assets.nasa.gov/image/PIA09000/PIA09000~thumb.jpg",
    "https://images-assets.nasa.gov/image/PIA09240/PIA09240~thumb.jpg",
    "https://images-assets.nasa.gov/image/PIA09853/PIA09853~thumb.jpg",
  ];

  it("renders and matches snapshot", () => {
    const { asFragment } = render(<SearchResults results={images} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays the expected number of search results", () => {
    const { getAllByTestId } = render(<SearchResults results={images} />);
    expect(getAllByTestId("image")).toHaveLength(7);
  });
});
