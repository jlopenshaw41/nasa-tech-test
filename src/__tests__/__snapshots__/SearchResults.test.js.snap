// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResults /> renders and matches snapshot 1`] = `
<DocumentFragment>
  <img
    alt="Search result from NASA database"
    class="image"
    data-testid="image"
    src="https://images-assets.nasa.gov/image/PIA06632/PIA06632~thumb.jpg"
  />
  <img
    alt="Search result from NASA database"
    class="image"
    data-testid="image"
    src="https://images-assets.nasa.gov/image/PIA06640/PIA06640~thumb.jpg"
  />
  <img
    alt="Search result from NASA database"
    class="image"
    data-testid="image"
    src="https://images-assets.nasa.gov/image/PIA07606/PIA07606~thumb.jpg"
  />
  <img
    alt="Search result from NASA database"
    class="image"
    data-testid="image"
    src="https://images-assets.nasa.gov/image/PIA08946/PIA08946~thumb.jpg"
  />
  <img
    alt="Search result from NASA database"
    class="image"
    data-testid="image"
    src="https://images-assets.nasa.gov/image/PIA09000/PIA09000~thumb.jpg"
  />
  <img
    alt="Search result from NASA database"
    class="image"
    data-testid="image"
    src="https://images-assets.nasa.gov/image/PIA09240/PIA09240~thumb.jpg"
  />
  <img
    alt="Search result from NASA database"
    class="image"
    data-testid="image"
    src="https://images-assets.nasa.gov/image/PIA09853/PIA09853~thumb.jpg"
  />
</DocumentFragment>
`;
